test/Main.purs

```diff
@@ -119,6 +120,25 @@ main = runTest do
         0.1 + 0.200001 ≇ 0.3
 
 
+  suite "fromString" do
+    test "valid number string" do
+      assert "integer strings are coerced" $
+        fromMaybe false $ map (_ == 123.0) $ fromString "123"
+
+      assert "decimals are coerced" $
+        fromMaybe false $ map (_ == 12.34) $ fromString "12.34"
+
+      assert "exponents are coerced" $
+        fromMaybe false $ map (_ == 1e4) $ fromString "1e4"
+
+      assert "decimals exponents are coerced" $
+        fromMaybe false $ map (_ == 1.2e4) $ fromString "1.2e4"
+
+    test "invalid number string" do
+      assert "invalid strings are not coerced" $
+        fromMaybe true $ false <$ fromString "bad string"
+
```

Comment:
Thanks for adding tests, too! Could you also add a test for an input like `1e1000`?